drivers/scheduler/eks/eks.go

```diff
@@ -0,0 +1,304 @@
+					break
+				}
+			}
+		}
+		if e.pxNodeGroupName == "" {
+			return fmt.Errorf("env EKS_PX_NODEGROUP_NAME or node label [%s] not found", nodeGroupLabel)
+		}
+	}
+	e.clusterName = os.Getenv("EKS_CLUSTER_NAME")
+	if e.clusterName == "" {
+		ec2InstanceLabel := "kubernetes.io/cluster/"
+		for _, node := range nodes.Items {
+			providerID := node.Spec.ProviderID
+			// In EKS, nodes have a ProviderID formatted as aws:///<region>/<instance-id>
+			splitID := strings.Split(providerID, "/")
+			if len(splitID) < 5 {
+				return fmt.Errorf("unexpected format of providerID: %s", providerID)
+			}
+			instanceID := splitID[4]
+			e.config, err = config.LoadDefaultConfig(context.TODO(), config.WithRegion(e.region))
+			if err != nil {
+				return fmt.Errorf("unable to load config for region %s, %v", e.region, err)
+			}
+			ec2Client := ec2.NewFromConfig(e.config)
+			result, err := ec2Client.DescribeInstances(
+				context.TODO(),
+				&ec2.DescribeInstancesInput{
+					InstanceIds: []string{instanceID},
+				},
+			)
+			if err != nil {
+				return fmt.Errorf("failed to describe instance %s, %v", instanceID, err)
+			}
+			for _, reservation := range result.Reservations {
+				for _, instance := range reservation.Instances {
+					for _, tag := range instance.Tags {
+						if strings.HasPrefix(*tag.Key, ec2InstanceLabel) {
+							e.clusterName = strings.TrimPrefix(*tag.Key, ec2InstanceLabel)
+							log.Infof("Instance [%s] is part of EKS cluster [%s] in region [%s]", instanceID, e.clusterName, e.region)
+							break
+						}
+					}
+				}
+			}
+		}
+		if e.clusterName == "" {
+			return fmt.Errorf("env EKS_CLUSTER_NAME or EC2 instance label [%s] not found", ec2InstanceLabel)
+		}
+	}
+	e.eksClient = eks.NewFromConfig(e.config)
+	return nil
+}
+
+// GetCurrentVersion returns the current version of the EKS cluster
+func (e *EKS) GetCurrentVersion() (string, error) {
+	eksDescribeClusterOutput, err := e.eksClient.DescribeCluster(
+		context.TODO(),
+		&eks.DescribeClusterInput{
+			Name: aws.String(e.clusterName),
+		},
+	)
+	if err != nil {
+		return "", fmt.Errorf("failed to describe EKS cluster [%s], Err: [%v]", e.clusterName, err)
+	}
+	if eksDescribeClusterOutput.Cluster == nil {
+		return "", fmt.Errorf("failed to describe EKS cluster [%s], cluster not found", e.clusterName)
+	}
+	return aws.ToString(eksDescribeClusterOutput.Cluster.Version), nil
+}
+
+// UpgradeControlPlane upgrades the EKS control plane to the specified version
+func (e *EKS) UpgradeControlPlane(version string) error {
+	log.Infof("Upgrade EKS Control Plane to version [%s]", version)
+	_, err := e.eksClient.UpdateClusterVersion(
+		context.TODO(),
+		&eks.UpdateClusterVersionInput{
+			Name:    aws.String(e.clusterName),
+			Version: aws.String(version),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to set cluser [%s] version to [%s], Err: [%v]", e.clusterName, version, err)
```

Comment:
Let's specify that its Control Plane cluster, like you have everywhere else

Reply:
Done.